src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/DfsServiceTests.cs

```diff
@@ -0,0 +1,306 @@
+        }
+
+        [Fact]
+        public async Task Can_Dispose()
+        {
+            using var node = TestDfs.GetTestDfs(_output);
+            await node.StartAsync();
+        }
+        
+        [Fact]
+        public async Task Can_Start_And_Stop()
+        {
+            var peer = _dfs1.LocalPeer;
+
+            Assert.False(_dfs1.IsStarted);
+            await _dfs1.StartAsync();
+            Assert.True(_dfs1.IsStarted);
+            Assert.NotEqual(0, peer.Addresses.Count());
+            await _dfs1.StopAsync();
+            Assert.False(_dfs1.IsStarted);
+            Assert.Equal(0, peer.Addresses.Count());
+
+            await _dfs1.StartAsync();
+            Assert.NotEqual(0, peer.Addresses.Count());
+            await _dfs1.StopAsync();
+            Assert.Equal(0, peer.Addresses.Count());
+
+            await _dfs1.StartAsync();
+            Assert.NotEqual(0, peer.Addresses.Count());
+            ExceptionAssert.Throws<Exception>(() => _dfs1.StartAsync().Wait());
+            await _dfs1.StopAsync();
+            Assert.Equal(0, peer.Addresses.Count());
+        }
+
+        [Fact]
+        public async Task Can_Start_And_Stop_MultipleEngines()
+        {
+            var peer1 = _dfs1.LocalPeer;
+            var peer2 = _dfs2.LocalPeer;
+
+            for (int n = 0; n < 3; ++n)
+            {
+                await _dfs1.StartAsync();
+                Assert.NotEqual(0, peer1.Addresses.Count());
+                await _dfs2.StartAsync();
+                Assert.NotEqual(0, peer2.Addresses.Count());
+
+                await _dfs2.StopAsync();
+                Assert.Equal(0, peer2.Addresses.Count());
+                await _dfs1.StopAsync();
+                Assert.Equal(0, peer1.Addresses.Count());
+            }
+        }
+
+        [Fact]
+        public async Task Can_Use_Private_Node()
+        {
+            using (var ipfs = TestDfs.GetTestDfs(_output))
+            {
+                ipfs.Options.Discovery.BootstrapPeers = new MultiAddress[0];
+                ipfs.Options.Swarm.PrivateNetworkKey = new PreSharedKey().Generate();
+                await _dfs1.StartAsync();
+            }
+        }
+
+        [Fact]
+        public async Task LocalPeer()
+        {
+            Task<Peer>[] tasks =
+            {
+                Task.Run(() => _dfs1.LocalPeer),
+                Task.Run(() => _dfs1.LocalPeer)
+            };
+            var r = await Task.WhenAll(tasks);
+            Assert.Equal(r[0], r[1]);
+        }
+
+        [Fact]
+        public async Task KeyChain()
+        {
+            Task<IKeyApi>[] tasks =
+            {
+                Task.Run(() => _dfs1.KeyApi),
+                Task.Run(() => _dfs1.KeyApi)
+            };
+            var r = await Task.WhenAll(tasks);
+            Assert.Equal(r[0], r[1]);
+        }
+
+        //todo
+        //[Fact]
+        //public async Task KeyChain_GetKey()
+        //{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)